### PR TITLE
Webpack scope hoisting

### PR DESCRIPTION
**Pull request by IgorMinar**

### CODE REVIEW

**Comment by IgorMinar** [comment]
I'm not quite sure why this is failing on CI. @petebacondarwin or @gkalpak can you please confirm that you can repro the error locally on your machines?

**Review by petebacondarwin — COMMENTED** [review]
Looks like Travis is not happy: https://travis-ci.org/angular/angular/jobs/240681110#L994

**Comment by IgorMinar** [comment]
@petebacondarwin yes, see my comment above. I can't repro this issue locally, so I wonder if anyone else can.

**Comment by IgorMinar** [comment]
this PR shaves off almost 250KB pre-gzip critical path payload, which brings us into a reasonable size territory, so I'd like to get this working and merged for the launch.

**Comment by petebacondarwin** [comment]
I'll try locally now

**Comment by petebacondarwin** [comment]
It builds locally for me...

**Comment by petebacondarwin** [comment]
I deleted the cache and tried it again.

**Comment by petebacondarwin** [comment]
Still the same error. Perhaps @gkalpak can reproduce on his Ubuntu VM?

File: aio/package.json [hunk 1]
```diff
@@ -35,7 +35,7 @@
     "sw-manifest": "ngu-sw-manifest --dist dist --in ngsw-manifest.json --out dist/ngsw-manifest.json",
     "sw-copy": "cp node_modules/@angular/service-worker/bundles/worker-basic.min.js dist/",
     "postinstall": "node tools/cli-patches/patch.js",
-    "build-ie-polyfills": "webpack -p src/ie-polyfills.js src/generated/ie-polyfills.min.js"
+    "build-ie-polyfills": "node_modules/webpack/bin/webpack.js -p src/ie-polyfills.js src/generated/ie-polyfills.min.js"
```
Comment (gkalpak) [inline aio/package.json]: This is not very Windows friendly 😞 Maybe `node node_modules/webpack/bin/webpack.js ...`?

**Comment by gkalpak** [comment]
Now it fails because of old Chromium version.

**Comment by gkalpak** [comment]
BTW, I can reproduce the issue locally. I guess it is because if the `PATCH_LOCK` file already exists, the new patches are not applied. We need `PATCH_LOCK` versioning (or delete node_modules every time the patches get updated) :smiley:

**Comment by gkalpak** [comment]
(I have been posting comments on wrong issues - Sorry)
So, yes I can reproduce it on Ubuntu. Not sure what is causing it yet.

**Comment by gkalpak** [comment]
FWIW, the error also happens without the patches, so I assume it is the updated cli (or some transient dependency updated through that).

**Comment by IgorMinar** [comment]
@gkalpak mentioned that this might be a bug in yarn that causes wrong typescript version to be provided to ngc/ngtools. Upgrading our tsc to 2.3 or yarn should resolve the problem. Since we upgraded to tsc 2.3 a bit earlier today I've rebased my PR on top of master to see if that resolves the issue.

**Comment by gkalpak** [comment]
>@gkalpak mentioned that this might be a bug in yarn that causes wrong typescript version to be provided to ngc/ngtools.

For future reference, it is not a bug in yarn. It is a difference in behavior between different minor yarn versions, which is expected, since yarn is still at v0.x. ([Some more details][1]).

[1]: https://github.com/angular/angular/pull/17350#issuecomment-307343091

**Pull request merged**